saos-webapp/src/test/java/pl/edu/icm/saos/enrichment/upload/EnrichmentTagUploadControllerTest.java

```diff
@@ -0,0 +1,398 @@
+    	
+    	// given
+    	long maxUploadSize = enrichmentTagUploadService.getEnrichmentTagMaxUploadSize();
+    	enrichmentTagUploadService.setEnrichmentTagMaxUploadSize(10);
+    	
+    	// execute 
+        ResultActions result = performPut("["+createJsonTag(12, "REFERENCED_REGULATIONS", "{'CCC':'YYY'}")+"]");
+        
+        enrichmentTagUploadService.setEnrichmentTagMaxUploadSize(maxUploadSize);
+        
+    	// assert
+        assertError(result, HttpStatus.REQUEST_ENTITY_TOO_LARGE, ERROR_MAX_UPLOAD_SIZE_EXCEEDED);
+        
+    }
+    
+    
+    @Test
+    public void uploadEnrichmentTags_SameTagAlreadyUploaded() throws Exception {
+    	
+    	// execute 
+        ResultActions result = performPut("["+createJsonTag(12, "REFERENCED_REGULATIONS", "{'XXX':'YYY'}")+","
+        									  +createJsonTag(12, "REFERENCED_REGULATIONS", "{'ABC':'YYY'}")
+        							     +"]");
+        
+        // assert
+        assertError(result, HttpStatus.BAD_REQUEST, ERROR_SAME_TAG_ALREADY_UPLOADED);
+        
+        assertEquals(0, enrichmentTagRepository.count());
+        assertEquals(0, uploadEnrichmentTagRepository.count());
+    }
+    
+    
+    
+    @Test
+    public void uploadEnrichmentTags_InternalServerError() throws Exception {
+    	
+    	// given
+    	EnrichmentTagUploadService mockEnrichmentTagUploadService = mock(EnrichmentTagUploadService.class);
+    	Mockito.doThrow(Exception.class).when(mockEnrichmentTagUploadService).uploadEnrichmentTags(Mockito.any(InputStream.class));
+    	enrichmentTagUploadController.setEnrichmentTagUploadService(mockEnrichmentTagUploadService);
+    	
+    	// execute 
+        ResultActions result = performPut("["+createJsonTag(12, "REFERENCED_REGULATIONS", "{'CCC':'YYY'}")+"]");
+        
+    	enrichmentTagUploadController.setEnrichmentTagUploadService(enrichmentTagUploadService);
+        
+    	// assert
+        assertError(result, HttpStatus.INTERNAL_SERVER_ERROR, EnrichmentTagUploadResponseMessages.ERROR_INTERVAL_SERVER_ERROR);
+    }
+    
+    
+    
+    @Test
+    public void uploadEnrichmentTags_OK() throws Exception {
+    	
+    	// given
+    	
+    	Judgment ccJudgment = testPersistenceObjectFactory.createCcJudgment();
+    	Judgment scJudgment = testPersistenceObjectFactory.createScJudgment();
+    	
+    	String jsonValue1 = normalizeJson("{'XXX':'YYY'}");
+    	String jsonValue2 = normalizeJson("{'caseNumbers':['YYY','xx-345']}");
+    	String jsonValue3 = normalizeJson("{'caseNumbers':['YYY','xx-345','YYY-2222']}");
+    	
+    	
+    	// execute 
+        
+    	ResultActions result = performPut("["+createJsonTag(ccJudgment.getId(), "REFERENCED_REGULATIONS", jsonValue1)+","
+   									 +createJsonTag(ccJudgment.getId(), "REFERENCED_CASE_NUMBERS", jsonValue2)+","
+   									 +createJsonTag(scJudgment.getId()+ccJudgment.getId(), "REFERENCED_CASE_NUMBERS", jsonValue2)+"," // non-existing judgment, should be copied to production tags
```

Comment:
It would be a little less confusing if `scJudgment.getId()+ccJudgment.getId()` would be moved to some variable.